lib/ansible/modules/cloud/atomic/atomic_container.py

```diff
@@ -0,0 +1,218 @@
+    type: string
+    sample: [u'Using default tag: latest ...']
+'''
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+
+
+def do_install(module, mode, rootfs, container, image, values_list, backend):
+    system_list = ["--system"] if mode == 'system' else []
+    user_list = ["--user"] if mode == 'user' else []
+    rootfs_list = ["--rootfs=%s" % rootfs] if rootfs else []
+    args = ['atomic', 'install', "--storage=%s" % backend, '--name=%s' % container] + system_list + user_list + rootfs_list + values_list + [image]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    else:
+        changed = "Extracting" in out or "Copying blob" in out
+        module.exit_json(msg=out, changed=changed)
+
+
+def do_update(module, container, image, values_list):
+    args = ['atomic', 'containers', 'update', "--rebase=%s" % image] + values_list + [container]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    else:
+        changed = "Extracting" in out or "Copying blob" in out
+        module.exit_json(msg=out, changed=changed)
+
+
+def do_uninstall(module, name, backend):
+    args = ['atomic', 'uninstall', "--storage=%s" % backend, name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    module.exit_json(msg=out, changed=True)
+
+
+def do_rollback(module, name):
+    args = ['atomic', 'containers', 'rollback', name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    else:
+        changed = "Rolling back" in out
+        module.exit_json(msg=out, changed=changed)
+
+
+def core(module):
+    mode = module.params['mode']
+    name = module.params['name']
+    image = module.params['image']
+    rootfs = module.params['rootfs']
+    values = module.params['values']
+    backend = module.params['backend']
+    state = module.params['state']
+
+    module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C')
+    out = {}
+    err = {}
+    rc = 0
+
+    values_list = ["--set=%s" % x for x in values] if values else []
+
+    args = ['atomic', 'containers', 'list', '--no-trunc', '-n', '--all', '-f', 'backend=%s' % backend, '-f', 'container=%s' % name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+        return
+    present = name in out
+
+    if state == 'present' and present:
+        module.exit_json(msg=out, changed=False)
+    elif (state in ['latest', 'present']) and not present:
+        do_install(module, mode, rootfs, name, image, values_list, backend)
+    elif state == 'latest':
+        do_update(module, name, image, values_list)
+    elif state == 'absent':
+        if not present:
+            module.exit_json(msg="", changed=False)
```

Comment:
A message string would good to say that image is not preset or something similar.